components/clp-mcp-server/clp_mcp_server/server/server.py

```diff
@@ -67,4 +72,32 @@ def hello_world(name: str = "clp-mcp-server user") -> dict[str, Any]:
             "status": "running",
         }
 
+    @mcp.tool
+    async def search_kql_query(kql_query: str, ctx: Context) -> dict[str, object]:
+        """
+        Searches the logs for the specified KQL query and returns the first page of the matching
+        result. The paginated results are ordered from latest to oldest, with each page containing
+        the paging metadata and 10 log messages along with their date string timestamps.
+
+        :param kql_query:
+        :param ctx: The `FastMCP` context containing the metadata of the underlying MCP session.
+        :return: Forwards `FastMCP.tool.get_nth_page`'s return values on success:
```

Comment:
```suggestion
        :return: A dictionary containing the following key-value pairs on success:
            - "items": A list of log entries in the requested page.
            - "num_total_pages": Total number of pages available from the query as an integer.
            - "num_total_items": Total number of log entries available from the query as an integer.
            - "num_items_per_page": Number of log entries per page.
            - "has_next": Whether a page exists after the returned one.
            - "has_previous": Whether a page exists before the returned one.
```
This isn't exactly correct. It doesn't forward `get_nth_page`, right?
I think it's better to explicitly write it out (to not confuse AI), and document in `get_page_data` (we documented in `NOTE` section that they should sync).

Reply:
I see, I will follow the same format as what you have wrote for get_neth_page.

This is a question for next pr where we implement substring_wz_timestamp, the type of the returned result is the same, how should I write the docstring there? Following your logic in not confusing AI, I think even though substring_wz_timestamp has the same return type with substring search, each return type should be listed explicitly.